### PR TITLE
Falcon - v1.0 -> Python and C

### PR DESCRIPTION
**Pull request by mrrostam**
Falcon is ready to be released! only need to release the python package after gettign approval

### CODE REVIEW

File: binding/python/README.md [hunk 1]
```diff
@@ -0,0 +1,56 @@
+# Falcon Speaker Diarization Engine
+
```
Comment (laves) [inline binding/python/README.md:2]: We're a bit inconsistent with the binding headers, but most of our SDKs have the format:
```markdown
# Falcon Binding for Python

## Falcon Speaker Diarization Engine
```

File: README.md [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+
+AccessKey is your authentication and authorization token for deploying Picovoice SDKs, including Falcon. Anyone who is
+using Picovoice needs to have a valid AccessKey. You must keep your AccessKey secret. You would need internet
+connectivity to validate your AccessKey with Picovoice license servers even though the speaker recognition is running
+100% offline.
+
+AccessKey also verifies that your usage is within the limits of your account. Everyone who signs up for
+[Picovoice Console](https://console.picovoice.ai/) receives the `Free Tier` usage rights described
+[here](https://picovoice.ai/pricing/). If you wish to increase your limits, you can purchase a subscription plan.
+
```
Comment (laves) [inline README.md:52]: Might be good to have a `What is Speaker Diarization?` section. People who really know what they are looking for will already know, but I do think it's a pretty technical term for the average user.

File: binding/python/README.md [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+## AccessKey
+
+Falcon requires a valid Picovoice `AccessKey` at initialization. `AccessKey` acts as your credentials when using Falcon SDKs.
+You can get your `AccessKey` for free. Make sure to keep your `AccessKey` secret.
+Signup or Login to [Picovoice Console](https://console.picovoice.ai/) to get your `AccessKey`.
+
+### Usage
+
+Create an instance of the engine and diarize an audio file:
```
Comment (laves) [inline binding/python/README.md:31]:
```suggestion
Create an instance of the engine and diarize an audio file (i.e. segment based on who is speaking):
```

File: binding/python/_falcon.py [hunk 4]
```diff
@@ -0,0 +1,335 @@
...
+        """Audio sample rate accepted by `.process`."""
+
+        return self._sample_rate
+
+    def _get_error_stack(self) -> Sequence[str]:
+        message_stack_ref = POINTER(c_char_p)()
+        message_stack_depth = c_int()
+        status = self._get_error_stack_func(byref(message_stack_ref), byref(message_stack_depth))
+        if status is not self.PicovoiceStatuses.SUCCESS:
+            raise self._PICOVOICE_STATUS_TO_EXCEPTION[status](message="Unable to get Porcupine error state")
```
Comment (laves) [inline binding/python/_falcon.py:310]:
```suggestion
            raise self._PICOVOICE_STATUS_TO_EXCEPTION[status](message="Unable to get Falcon error state")
```

File: .github/workflows/python.yml [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+      - 'binding/python/**'
+      - '!binding/python/README.md'
+      - 'lib/common/**'
+      - 'lib/jetson/**'
+      - 'lib/linux/**'
+      - 'lib/mac/**'
+      - 'lib/raspberry-pi/**'
+      - 'lib/windows/**'
+      - 'resources/audio_samples/*.wav'
+  pull_request:
```
Comment (laves) [inline .github/workflows/python.yml:18]: add `resources/.test/**`

File: include/pv_falcon.h [hunk 6]
```diff
@@ -0,0 +1,150 @@
...
+ *         - `PV_STATUS_ACTIVATION_REFUSED`: Activation refused.
+ */
+PV_API pv_status_t pv_falcon_process_file(
+        pv_falcon_t *object,
+        const char *audio_path,
+        int32_t *num_segments,
+        pv_segment_t **segments);
+
+/**
+ * Deletes segments allocated by `pv_falcon_process()`.
```
Comment (laves) [inline include/pv_falcon.h:128]:
```suggestion
 * Deletes segments allocated by `pv_falcon_process()` and `pv_falcon_process_file()`.
```

File: include/pv_falcon.h [hunk 7]
```diff
@@ -0,0 +1,150 @@
...
+ *
+ * Use this function to properly release memory allocated for segments returned by the
+ * `pv_falcon_process()` functions.
```
Comment (laves) [inline include/pv_falcon.h:131]:
```suggestion
 * `pv_falcon_process()` and `pv_falcon_process_file()` functions.
```

File: README.md (hunk 2 above)
Comment (mrrostam) [reply README.md:58]: Added:
```
## What is Speaker Diarization?

Speaker diarization, a fundamental step in automatic speech recognition and audio processing, focuses on identifying and
separating distinct speakers within an audio recording. Its objective is to divide the audio into segments while
precisely identifying the speakers and their respective speaking intervals.

```

File: binding/python/_falcon.py (hunk 4 above)
Comment (mrrostam) [reply binding/python/_falcon.py]: good catch!

**Review by ErisMik — APPROVED** [review]

File: .github/workflows/python.yml [hunk 8]
```diff
@@ -28,6 +28,7 @@ on:
       - 'lib/raspberry-pi/**'
       - 'lib/windows/**'
       - 'resources/audio_samples/*.wav'
+      - 'resources/.test/**'
```
Comment (laves) [inline .github/workflows/python.yml:31]: should add to the `push:` list as well

**Review by laves — CHANGES_REQUESTED** [review]

**Review by laves — APPROVED** [review]

**Pull request merged**